### PR TITLE
bump bundler-head to 1.8.9

### PR DESCRIPTION
**Pull request by phunehehe**
see https://github.com/NixOS/nixpkgs/issues/12785

### CODE REVIEW

**Comment by phunehehe** [comment]
(I'm not entirely sure what the failed check is about, looks like Travis failed the build because it logs too much?)
Alternatively, remove `leaveDotGit = true`. Is it useful for anything other than debugging?

**Comment by fpletz** [comment]
It doesn't make a lot of sense to bump bundler-head, which I guess is supposed to track the development version from the bundler master to a version that is even older then the current version of bundler in nixpkgs which is 1.10.5, see https://github.com/phunehehe/nixpkgs/blob/9dbef54378811bf3b95b41b17e3691ed7e428db7/pkgs/development/interpreters/ruby/bundler.nix#L4.

I'm not sure why .git is need but probably to generate version with the commit hash. You could try to fix/patch that if something fails.

**Comment by phunehehe** [comment]
I explained why in https://github.com/NixOS/nixpkgs/issues/12785

> To fight the non-deterministic fetchgit, I propose replacing bundler/bundler@a2343c9 with the latest minor release of the oldest major release containing the commit, which is v1.8.9.

It might have been the case back then, to create bundler-head tracking a commit that we need, that has not been released. When that commit is released, bundler-head should be replaced with the release so that we are back to a normal release. This PR just finishes that missing step.

The only reason I see not to do that is backward compatibility. However, because this is broken (non-deterministic hashes causing builds to fail), I'm pretty sure no one can use it, and so it's not actually a backward compatibility problem.

File: pkgs/development/interpreters/ruby/bundler-head.nix [hunk 1]
```diff
@@ -1,13 +1,9 @@
...
-  name = "bundler-2015-01-11";
-  src = fetchgit {
-    url = "https://github.com/bundler/bundler.git";
-    rev = "a2343c9eabf5403d8ffcbca4dea33d18a60fc157";
-    sha256 = "1p7kzhmicfljy9n7nq3qh6lvrsckiq76ddypf6s55gfh1l98z4k9";
-    leaveDotGit = true;
-  };
+  name = "bundler-1.8.9";
+  namePrefix = "";
+  sha256 = "1k4sk4vf0mascqnahdnqymhr86dqj92bddciz5b2p9sv3qzryq57";
```
Comment (jagajaga) [inline pkgs/development/interpreters/ruby/bundler-head.nix]: You removed src fetching.

Comment (phunehehe) [inline pkgs/development/interpreters/ruby/bundler-head.nix]: yep, this works the same way as https://github.com/NixOS/nixpkgs/blob/release-15.09/pkgs/development/interpreters/ruby/bundler.nix

**Comment by fpletz** [comment]
Thanks for the explanation & the fix. That makes sense! :+1:

**Comment by phunehehe** [comment]
\o/

**Pull request merged**